Spanner/src/SpannerClient.php

```diff
@@ -325,6 +376,44 @@ public function instance($name, array $instance = [])
             $this->returnInt64AsObject,
             $instance
         );
+
+        if (!isset($this->config['apiEndpoint']) &&
+            getenv('GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING')
+        ) {
+            if (!isset($this->endpoints[$instanceObject->name()])) {
+                $instanceInfo = $instanceObject->info([
+                    'fieldMask' => new FieldMask(['paths' => ['endpoint_uris']])
+                ]);
```

Comment:
I'm afraid, this lines can break the instance. When we're setting `fieldMask`, we're requesting only fields mentioned in `fieldMask` while others will got default values (zeros and empty strings). So, this way:

`info()` calls `reload()`:
https://github.com/q-logic/google-cloud-php/blob/3e3189a7d15d66d8177d1115430133674af5864d/Spanner/src/Instance.php#L184-L187
which overrides `instance->info` parameter with the return from `getInstance()`:
https://github.com/q-logic/google-cloud-php/blob/3e3189a7d15d66d8177d1115430133674af5864d/Spanner/src/Instance.php#L234-L239
Thus, new `info` in our case is not empty, but consists of endpoints list and zero values for other fields, so future calls to instance probably will be broken, as we redefined all of it's (local object, not the one in the base) fields except endpoints list. Here is an example from the live service, I've only mocked the endpoints list:
![image](https://user-images.githubusercontent.com/26030071/70526153-0437e500-1b5a-11ea-8afb-f5652a83d995.png)

Reply:
You're right! I added all fields for instance to FieldMask. Checked the result on a small script. Changes are inserted into the next commit.
![FieldMask - is working](https://user-images.githubusercontent.com/15671241/70541598-5f78d000-1b78-11ea-8946-8e6455a63732.jpg)
Now, as you can see the result of running instance with FieldMask and with default value it will not change the values in the instance object itself.

Reply:
That's not good. If new field will be added, than someone will have to add this field into this list. And after all, filter with all fields sounds equal to absence of filter at all